wisp/wisp-rate-limiting/src/main/kotlin/wisp/ratelimiting/RateLimiter.kt

```diff
@@ -0,0 +1,78 @@
+package wisp.ratelimiting
+
+import java.io.IOException
+import java.time.Instant
+
+/**
+ * An interface for acquiring and releasing rate limit tokens
+ *
+ * Terms:
+ * * `key` - a unique identifier for the entity or operation being rate limited
+ *
+ * * `bucket` - a container for rate limit tokens applied to a specific key
+ *
+ * * `token` - a unit of rate limit capacity. When we go to perform a rate limited operation, we
+ * attempt to consume a token from the bucket. If successful, we can perform the operation.
+ *
+ * See also: https://en.wikipedia.org/wiki/Token_bucket
+ */
+interface RateLimiter {
+  /**
+   * Consumes [amount] tokens from the bucket associated with the given key
+   * TODO Confirm the exception type is correct
```

Comment:
Do we still need to confirm this?

Reply:
This is actually incorrect, it surfaces whatever exception the underlying library throws (e.g. `JedisException` subclasses). Will add a note about that in the doc string